File: server/Bank.Core/JsonModels/Bank/Customers/CustomerAccountSummaryModel.cs
using Bank.Core.Enums;

namespace Bank.Core.JsonModels.Bank.Customers;

public class CustomerAccountSummaryModel
{
    public long Id { get; set; }
    public string Iban { get; set; } = string.Empty;
    public decimal Balance { get; set; }
    public BankAccountStatus Status { get; set; }
    public DateTime OpenedAtUtc { get; set; }
    public DateTime? ClosedAtUtc { get; set; }
}

File: client/src/types/JsonModels/Bank/Customers/CustomerAccountSummaryModel.ts
import type { BankAccountStatus } from "../../Enums/BankAccountStatus";

export interface CustomerAccountSummaryModel
{
	id: number;
	iban: string;
	balance: number;
	status: BankAccountStatus;
	openedAtUtc: string;
	closedAtUtc?: string;
}
